src/System.Management.Automation/engine/NativeCommandParameterBinder.cs

```diff
@@ -324,12 +340,76 @@ private void PossiblyGlobArg(string arg, bool usedQuotes)
             }
 #endif // UNIX
 
+            // Check if the start of the path is a filesystem drive
+            if (stringConstantType != StringConstantType.SingleQuoted)
+            {
+                arg = ResolvePath(arg, Context);
+            }
+
             if (!argExpanded)
             {
                 _arguments.Append(arg);
             }
         }
 
+        /// <summary>
+        /// Check if string is prefixed by psdrive, if so, expand it if filesystem path.
+        /// </summary>
+        /// <param name="path">The potential PSPath to resolve.</param>
+        /// <param name="context">The current ExecutionContext.</param>
+        /// <returns>Resolved PSPath if applicable otherwise the original path</returns>
+        internal static string ResolvePath(string path, ExecutionContext context)
+        {
+            if (ExperimentalFeature.IsEnabled("PSNativePSPathResolution"))
+            {
+                string driveName;
+#if !UNIX
+                if (string.Equals(path, "~") || path.StartsWith("~" + Path.DirectorySeparatorChar) || path.StartsWith("~" + Path.AltDirectorySeparatorChar))
+                {
+                    try
+                    {
+                        ProviderInfo fileSystemProvider = context.EngineSessionState.GetSingleProvider(FileSystemProvider.ProviderName);
+                        return (fileSystemProvider.Home + path.Substring(1)).Replace(Path.AltDirectorySeparatorChar, Path.DirectorySeparatorChar);
+                    }
+                    catch
+                    {
+                        return path;
+                    }
+                }
+                else
+#endif
+                if (context.SessionState.Path.IsPSAbsolute(path, out driveName))
+                {
+#if !UNIX
+                    // check if the driveName is an actual disk drive on Windows, if so, no expansion
+                    foreach (var drive in DriveInfo.GetDrives())
+                    {
+                        if (drive.Name.StartsWith(driveName))
+                        {
+                            return path;
+                        }
+                    }
+#endif
+                    try
+                    {
+                        ProviderInfo provider;
+                        Collection<string> paths = context.SessionState.Path.GetResolvedProviderPathFromPSPath($"{driveName}:", out provider);
+                        if (paths.Count == 1 && string.Equals(provider.Name, FileSystemProvider.ProviderName))
+                        {
+                            // + 2 to replace the colon and the trailing slash which is part of the returned pspath
+                            return (paths[0] + path.Substring(driveName.Length + 2)).Replace(Path.AltDirectorySeparatorChar, Path.DirectorySeparatorChar);
```

Comment:
```suggestion
                            return new StringBuilder(paths[0])
                                .Append(path.Substring(driveName.Length + 2))
                                .Replace(Path.AltDirectorySeparatorChar, Path.DirectorySeparatorChar)
                                .ToString();
```

With same comment about replacing the path separator with itself on unix

Reply:
Will update.  Can enclose the .Replace with `#if !UNIX`